client/data/payment-intents/hooks.ts

```diff
@@ -10,7 +10,7 @@ import { ChargeResponse } from '../charges/types';
 import { STORE_NAME } from '../constants';
 
 export const getIsChargeId = ( id: string ): boolean =>
-	-1 !== id.indexOf( 'ch_' );
+	-1 !== id.indexOf( 'ch_' ) || -1 !== id.indexOf( 'py_' );
```

Comment:
Should we still use this validation, or remove it like in the server?

Reply:
We need this in the client to perform the fallback behavior of loading the transaction details from the charge id. It will fetch the payment intent data from the charge id and then redirect to the equivalent `pi_` id.

Reply:
We could rely on the payment intent id and assume that if it doesn't include `pi_` then it's a charge id.

Reply:
In this case, newer orders shouldn't generate the `charge_id` links, as everything will switch to using the payment intent ID. The check here is to provide the fallback to old links and we need it to identify if it is using an intent or charge to request the page.